python_modules/dagster/dagster/core/asset_defs/asset_collection.py

```diff
@@ -0,0 +1,107 @@
+from typing import Dict, List, NamedTuple, Optional
+
+from dagster import check
+from dagster.utils import merge_dicts
+
+from ..definitions.executor_definition import ExecutorDefinition
+from ..definitions.resource_definition import ResourceDefinition
+from ..errors import DagsterInvalidDefinitionError
+from .asset import AssetsDefinition
+from .assets_job import build_root_manager, build_source_assets_by_key
+from .foreign_asset import ForeignAsset
+
+
+class AssetCollection(
+    NamedTuple(
+        "_AssetCollection",
+        [
+            ("assets", List[AssetsDefinition]),
+            ("source_assets", List[ForeignAsset]),
+            ("resource_defs", Dict[str, ResourceDefinition]),
+            ("executor_def", Optional[ExecutorDefinition]),
+        ],
+    )
+):
+    def __new__(
+        cls,
+        assets: List[AssetsDefinition],
+        source_assets: List[ForeignAsset],
```

Comment:
source & foreign ?

Reply:
I think we know that we want to move away from the name foreign, this was preemptive though